### PR TITLE
[FLINK-2642] [table] Scala Table API crashes when executing word count example

### PR DESCRIPTION
**Pull request by twalthr**
This PR improves the checking of the types during table translation. The error message for [FLINK-2642] is now correct.

### CODE REVIEW

**Comment by aljoscha** [comment]
looks good +1 :smile:

File: flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/plan/PlanTranslator.scala [hunk 1]
```diff
@@ -110,7 +110,8 @@ abstract class PlanTranslator {
     }
 
     val clazz = repr.getType().getTypeClass
-    if (clazz.isMemberClass && !Modifier.isStatic(clazz.getModifiers)) {
+    if ((clazz.isMemberClass && !Modifier.isStatic(clazz.getModifiers))
+        || clazz.getCanonicalName() == null) {
```
Comment (hsaputra) [inline flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/plan/PlanTranslator.scala]: Gah, another Java problem of returning null =(

**Comment by hsaputra** [comment]
+1

**Comment by aljoscha** [comment]
@twalthr Please go ahead and merge it. :smile: Then we can close the issue.

**Pull request closed without merging**